api/src/services/applications/get_application_form.py

```diff
@@ -4,20 +4,24 @@
 
 import src.adapters.db as db
 from src.api.route_utils import raise_flask_error
-from src.db.models.competition_models import Application, ApplicationForm
+from src.db.models.competition_models import ApplicationForm
+from src.db.models.user_models import User
 from src.form_schema.jsonschema_validator import (
     ValidationErrorDetail,
     validate_json_schema_for_form,
 )
+from src.services.applications.auth_utils import check_user_application_access
+from src.services.applications.get_application import get_application
 
 
 def get_application_form(
-    db_session: db.Session, application_id: UUID, app_form_id: UUID
+    db_session: db.Session, application_id: UUID, app_form_id: UUID, user: User
 ) -> tuple[ApplicationForm, list[ValidationErrorDetail]]:
-    # Check if application exists
-    application = db_session.execute(
-        select(Application).where(Application.application_id == application_id)
-    ).scalar_one_or_none()
+    """
+    Get an application form by ID, checking if the user has access to it.
+    """
+    # Get the application
+    application = get_application(db_session, application_id, user)
 
     if not application:
```

Comment:
Since we have this 404 logic in the get_application function, could remove it here